chain/client/src/client.rs

```diff
@@ -111,6 +114,11 @@ pub struct Client {
     /// Last time the head was updated, or our head was rebroadcasted. Used to re-broadcast the head
     /// again to prevent network from stalling if a large percentage of the network missed a block
     last_time_head_progress_made: Instant,
+
+    /// Block and chunk production timing information.
+    /// used only for debug purposes.
+    pub block_production_times: lru::LruCache<BlockHeight, BlockProduction>,
+    pub chunk_production_times: lru::LruCache<(BlockHeight, ShardId), Duration>,
```

Comment:
Hm, using LRU caches here seems a bit weired, I'd expect to see a VecDeque setup here instead.

Reply:
The logic is a little bit different than in doomslug approvals sent -- here I need a map (as for debugging, I'm looking up production at a given height).

LRU is not perfect here -- I'd like to have 'least recently INSERTED' - but I couldn't find it - so I've used LRU instead.

Reply:
https://docs.rs/indexmap/latest/indexmap/ might be the thing you are looking for. indexmap is a good crate.